### PR TITLE
Add recipe for systemd-python

### LINKED ISSUE #6086: Package systemd-python
It would be nice to have a package for [systemd-python](https://pypi.org/project/systemd-python), not to be confused with [python-systemd](https://pypi.org/project/python-systemd). `systemd-python` is a python module for native access to the systemd facilities. 

cf: #6084

### PR DESCRIPTION
**Pull request by CurtLH**
Closes #6086

### CODE REVIEW

**Comment by CurtLH** [comment]
I'm getting the following error when the package is attempting to build:

```bash
Processing /home/conda/feedstock_root/build_artifacts/systemd-python_1528941666810/work
    Complete output from command python setup.py egg_info:
    Cannot find libsystemd or libsystemd-journal:
    
    Package libsystemd was not found in the pkg-config search path.
    Perhaps you should add the directory containing `libsystemd.pc'
    to the PKG_CONFIG_PATH environment variable
    No package 'libsystemd' found
    
    Package libsystemd-journal was not found in the pkg-config search path.
    Perhaps you should add the directory containing `libsystemd-journal.pc'
    to the PKG_CONFIG_PATH environment variable
    No package 'libsystemd-journal' foundProcessing /home/conda/feedstock_root/build_artifacts/systemd-python_1528941666810/work
    Complete output from command python setup.py egg_info:
    Cannot find libsystemd or libsystemd-journal:
    
    Package libsystemd was not found in the pkg-config search path.
    Perhaps you should add the directory containing `libsystemd.pc'
    to the PKG_CONFIG_PATH environment variable
    No package 'libsystemd' found
    
    Package libsystemd-journal was not found in the pkg-config search path.
    Perhaps you should add the directory containing `libsystemd-journal.pc'
    to the PKG_CONFIG_PATH environment variable
    No package 'libsystemd-journal' found
```

Based on [this comment](https://github.com/systemd/python-systemd/issues/47#issuecomment-293945651), it seems that `systemd` must be installed first.

**Comment by hmaarrfk** [comment]
I think you need to add `pkg-config` in host (among other things probably)

File: recipes/systemd-python/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  noarch: python
+  number: 0
+  script: python -m pip install --no-deps --ignore-installed .
```
Comment (marcelotrevisani) [inline recipes/systemd-python/meta.yaml]: Please change that to

```
  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed --no-cache-dir -vvv"
```

File: recipes/systemd-python/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  noarch: python
```
Comment (marcelotrevisani) [inline recipes/systemd-python/meta.yaml]: This recipe can't be ``noarch`` because it seems you need the compilers right?

**Comment by jakirkham** [comment]
FWIW have raised issue ( https://github.com/conda-forge/staged-recipes/issues/8143 ) to package `systemd`.

**Pull request closed without merging**